Runtime to set default EntryPoint for [UnmanagedCallersOnly]

`EntryPoint` is a nullable field:
https://github.com/dotnet/runtime/blob/988f570dc4ddd79a166ca8cbc184b2938ed2fefc/src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/InteropServices/UnmanagedCallersOnlyAttribute.cs#L36

Omitting it in usage should fallback to the default value (method name on which it was decorated) by the runtime here:
https://github.com/dotnet/runtime/blob/988f570dc4ddd79a166ca8cbc184b2938ed2fefc/src/coreclr/vm/callconvbuilder.cpp#L527

to avoid compilation errors from native toolchain.

#### Repro

```sh
# bash on ubuntu
$ dotnet new classlib -n nativelib1 && cd nativelib1
$ cat > Class1.cs << EOF
using System;
using System.Runtime.InteropServices;

public class C
{
    [UnmanagedCallersOnly]
    public static int Sum (int x, int y) => x + y;
}
EOF

$ dotnet7 publish -o dist --use-current-runtime -c Release -p:PublishAot=true -p:NativeLib=shared
```
 
#### Expected

Exposes an entry point with name `Sum`.

#### Actual

Errors out:

```
   Generating compatible native code. To optimize for size or speed, visit https://aka.ms/OptimizeNativeAOT
  /usr/bin/ld:obj/Release/net7.0/linux-x64/native/nativelib1.exports:3: syntax error in VERSION script
collect2 : error : ld returned 1 exit status [/home/am11/projects/nativelib1/nativelib1.csproj]
/home/am11/.dotnet7/sdk/7.0.100-preview.7.22324.4/Sdks/Microsoft.DotNet.ILCompiler/build/Microsoft.NETCore.Native.targets(338,5): error MSB3073: The command "gcc "obj/Release/net7.0/linux-x64/native/nativelib1.o" -o "bin/Release/net7.0/linux-x64/native/nativelib1.so" -Wl,--version-script=obj/Release/net7.0/linux-x64/native/nativelib1.exports /home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.6.22322.7/sdk/libbootstrapperdll.a /home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.6.22322.7/sdk/libRuntime.WorkstationGC.a /home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.6.22322.7/framework/libSystem.Native.a /home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.6.22322.7/framework/libSystem.Globalization.Native.a /home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.6.22322.7/framework/libSystem.IO.Compression.Native.a /home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.6.22322.7/framework/libSystem.Net.Security.Native.a /home/am11/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/7.0.0-preview.6.22322.7/framework/libSystem.Security.Cryptography.Native.OpenSsl.a -g -Wl,-rpath,'$ORIGIN' -Wl,--build-id=sha1 -Wl,--as-needed -pthread -lstdc++ -ldl -lm -lz -lrt -shared -Wl,-z,relro -Wl,-z,now -Wl,--require-defined,NativeAOT_StaticInitialization -Wl,--discard-all -Wl,--gc-sections" exited with code 1. [/home/am11/projects/nativelib1/nativelib1.csproj]
```

The linker script has `;` instead of `Sum;`:

```sh
$ cat obj/Release/net7.0/linux-x64/native/nativelib1.exports
V1.0 {
    global: _init; _fini;
        ;
    local: *;
};
```

#### Workaround

Replace `[UnmanagedCallersOnly]` with `[UnmanagedCallersOnly(EntryPoint = nameof(Sum))]`

--- a/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportedMethodsRootProvider.cs
+++ b/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportedMethodsRootProvider.cs
@@ -29,9 +29,8 @@ namespace ILCompiler
                     foreach (var method in type.GetMethods())
                     {
                         EcmaMethod ecmaMethod = (EcmaMethod)method;
-                        if (ecmaMethod.IsRuntimeExport && ecmaMethod.GetRuntimeExportName() != null)
-                            yield return ecmaMethod;
-                        else if (ecmaMethod.IsUnmanagedCallersOnly && ecmaMethod.GetUnmanagedCallersOnlyExportName() != null)
+                        if ((ecmaMethod.IsRuntimeExport && ecmaMethod.GetRuntimeExportName() != null) ||
+                            (ecmaMethod.IsUnmanagedCallersOnly && ecmaMethod.GetUnmanagedCallersOnlyExportName() != null))
                             yield return ecmaMethod;
                     }
                 }

--- a/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportedMethodsRootProvider.cs
+++ b/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportedMethodsRootProvider.cs
@@ -29,13 +29,10 @@ namespace ILCompiler
                     foreach (var method in type.GetMethods())
                     {
                         EcmaMethod ecmaMethod = (EcmaMethod)method;
-                        if (ecmaMethod.IsRuntimeExport || ecmaMethod.IsUnmanagedCallersOnly)
-                        {
-                            if (ecmaMethod.IsRuntimeExport && ecmaMethod.GetRuntimeExportName() != null)
-                                yield return ecmaMethod;
-                            else if (ecmaMethod.IsUnmanagedCallersOnly && ecmaMethod.GetUnmanagedCallersOnlyExportName() != null)
-                                yield return ecmaMethod;
-                        }
+                        if (ecmaMethod.IsRuntimeExport && ecmaMethod.GetRuntimeExportName() != null)
+                            yield return ecmaMethod;
+                        else if (ecmaMethod.IsUnmanagedCallersOnly && ecmaMethod.GetUnmanagedCallersOnlyExportName() != null)
+                            yield return ecmaMethod;
                     }
                 }
             }

--- a/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportedMethodsRootProvider.cs
+++ b/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportedMethodsRootProvider.cs
@@ -30,7 +30,12 @@ namespace ILCompiler
                     {
                         EcmaMethod ecmaMethod = (EcmaMethod)method;
                         if (ecmaMethod.IsRuntimeExport || ecmaMethod.IsUnmanagedCallersOnly)
-                            yield return ecmaMethod;
+                        {
+                            if (ecmaMethod.IsRuntimeExport && ecmaMethod.GetRuntimeExportName() != null)
+                                yield return ecmaMethod;
+                            else if (ecmaMethod.IsUnmanagedCallersOnly && ecmaMethod.GetUnmanagedCallersOnlyExportName() != null)
+                                yield return ecmaMethod;
+                        }
                     }
                 }
             }
@@ -43,14 +48,12 @@ namespace ILCompiler
                 if (ecmaMethod.IsRuntimeExport)
                 {
                     string runtimeExportName = ecmaMethod.GetRuntimeExportName();
-                    if (runtimeExportName != null)
-                        rootProvider.AddCompilationRoot((MethodDesc)ecmaMethod, "Runtime export", runtimeExportName);
+                    rootProvider.AddCompilationRoot((MethodDesc)ecmaMethod, "Runtime export", runtimeExportName);
                 }
                 else if (ecmaMethod.IsUnmanagedCallersOnly)
                 {
                     string unmanagedCallersOnlyExportName = ecmaMethod.GetUnmanagedCallersOnlyExportName();
-                    if (unmanagedCallersOnlyExportName != null)
-                        rootProvider.AddCompilationRoot((MethodDesc)ecmaMethod, "Native callable", unmanagedCallersOnlyExportName);
+                    rootProvider.AddCompilationRoot((MethodDesc)ecmaMethod, "Native callable", unmanagedCallersOnlyExportName);
                 }
             }
         }

